app/sidero-controller-manager/internal/metadata/metadata_server.go

```diff
@@ -251,62 +249,102 @@ func patchConfigs(decodedData []byte, patches []metalv1.ConfigPatches) ([]byte,
+		return nil, errorWithCode{http.StatusInternalServerError, fmt.Errorf("failure loading rfc6902 patches from server: %s", err)}
 	}
 
-	jsonDecodedData, err = patch.Apply(jsonDecodedData)
+	patched, err := configpatcher.Apply(configpatcher.WithBytes(decodedData), []configpatcher.Patch{patch})
 	if err != nil {
 		return nil, errorWithCode{http.StatusInternalServerError, fmt.Errorf("failure applying rfc6902 patches to machine config: %s", err)}
 	}
 
-	decodedData, err = yaml.JSONToYAML(jsonDecodedData)
+	result, err := patched.Bytes()
 	if err != nil {
-		return nil, errorWithCode{http.StatusInternalServerError, fmt.Errorf("failure converting bootstrap data from json to yaml: %s", err)}
+		return nil, errorWithCode{http.StatusInternalServerError, fmt.Errorf("failure converting patched config to bytes: %s", err)}
 	}
 
-	return decodedData, errorWithCode{}
+	return result, errorWithCode{}
 }
 
 // labelNodes is responsible for editing the kubelet extra args such that a given
 // server gets registered with a label containing the UUID of the server resource it's actually running on.
 func labelNodes(decodedData []byte, serverName string) ([]byte, errorWithCode) {
-	configProvider, err := configloader.NewFromBytes(decodedData)
-	if err != nil {
+	// avoid using the `configloader` from Talos machinery here, as it will fail on "unknown" fields
+	// causing a dependency on Talos version that Sidero was built with
+	var cfg struct {
+		Version string `yaml:"version"`
+		Machine *struct {
+			Kubelet *struct {
+				ExtraArgs map[string]string `yaml:"extraArgs"`
+			} `yaml:"kubelet"`
+		} `yaml:"machine"`
+	}
+
+	if err := yaml.Unmarshal(decodedData, &cfg); err != nil {
 		return nil, errorWithCode{http.StatusInternalServerError, fmt.Errorf("failure creating config struct: %s", err)}
 	}
 
-	switch configProvider.Version() {
+	switch cfg.Version {
 	case "v1alpha1":
-		config, ok := configProvider.Raw().(*talosv1alpha1.Config)
-		if !ok {
-			return nil, errorWithCode{http.StatusInternalServerError, fmt.Errorf("unable to case config")}
-		}
+		var (
+			patch      metalv1.ConfigPatches
+			patchValue any
+		)
 
-		patch := metalv1.ConfigPatches{
-			Path: "/machine/kubelet/extraArgs",
-			Op:   "replace",
-		}
+		label := fmt.Sprintf("metal.sidero.dev/uuid=%s", serverName)
 
-		kubeletExtraArgs := config.MachineConfig.MachineKubelet.KubeletExtraArgs
-		if kubeletExtraArgs == nil {
-			patch.Op = "add"
-			kubeletExtraArgs = make(map[string]string)
-		}
+		switch {
+		case cfg.Machine == nil:
+			patch = metalv1.ConfigPatches{
+				Path: "/machine",
+				Op:   "add",
+			}
 
-		if _, ok = kubeletExtraArgs["node-labels"]; ok {
-			kubeletExtraArgs["node-labels"] += fmt.Sprintf(",metal.sidero.dev/uuid=%s", serverName)
-		} else {
-			kubeletExtraArgs["node-labels"] = fmt.Sprintf("metal.sidero.dev/uuid=%s", serverName)
+			patchValue = map[string]any{
+				"kubelet": map[string]any{
```

Comment:
machine.nodeLabels is better?

Reply:
it is, but not supported on all versions of Talos, I don't want to lock in to >= 1.3